### PR TITLE
LuaJIT: update to 2.1.1713773202, cleanup, adopt.

### PR DESCRIPTION
**Pull request by Calandracas606**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64-musl
  - i686
  - aarch64 (cross)
  - aarch64-musl (cross)
  - armv7l (cross)
  - armv7l-musl (cross)
  - armv6l (cross)
  - armv6l-musl (cross)

### CODE REVIEW

**Comment by Calandracas606** [comment]
based on https://github.com/void-linux/void-packages/pull/48453 with some additional changes

File: srcpkgs/LuaJIT/template [hunk 1]
```diff
@@ -1,71 +1,55 @@
...
+	if [ "$build_option_lua52compat" ]; then
+		make_build_args+=" XCFLAGS=-DLUAJIT_ENABLE_LUA52COMPAT"
+	fi
 }
 
 do_build() {
+	# if we don't unset, the build fails to cross-compile
+	# due to confliction with the makefile macros
 	local _cflags=$CFLAGS
 	local _ldflags=$LDFLAGS
```
Comment (classabbyamp) [inline srcpkgs/LuaJIT/template:43]:
```suggestion
	local _cflags="$CFLAGS"
	local _ldflags="$LDFLAGS"
```

**Comment by xolophreny** [comment]
Hi, would like this to be merged, if possible

**Comment by AnInternetTroll** [comment]
Tested with neovim (x86_64) and works as expected, didn't notice any bugs.

**Pull request merged**